src/aiovodafone/api.py

```diff
@@ -39,6 +39,34 @@ class VodafoneStationDevice:
 class VodafoneStationCommonApi(ABC):
     """Common API calls for Vodafone Station routers."""
 
+    @staticmethod
+    async def get_device_type(
+        host: str, session: aiohttp.ClientSession
+    ) -> DeviceType | None:
+        """Finds out the device type of a Vodafone Stations and returns it as enum.
+        The Technicolor devices always answer with a valid HTTP response, the
+        Sercomm returns 404 on a missing page. This helps to determine which we are
+        talking with.
+
+        Args:
+            host (str): The router's address, e.g. `192.168.1.1`
+            session (aiohttp.ClientSession): the client session to issue HTTP request with
+
+        Returns:
+            DeviceType: If the device is a Technicolor, it returns
+            `DeviceType.TECHNICOLOR`. If the device is a Sercomm, it returns `DeviceType.SERCOMM`.
+            If neither of the device types match, it returns `None`.
+        """
+        async with session.get(f"http://{host}/api/v1/login_conf") as response:
+            if response.status == 200:
+                response_json = await response.json()
+                if "data" in response_json and "ModelName" in response_json["data"]:
+                    return DeviceType.TECHNICOLOR
+        async with session.get(f"http://{host}/login.html") as response:
```

Comment:
A lot of devices return a login.html. I would also check for at least `csrf_token` like in `_get_csrf_token()` (making this a generic method so to avoid code duplication.

Reply:
Good point. The `_get_csrf_token()` right now is only needed in the Sercomm implementation. For the static method to use this, we would even have to convert it into a static method as well. Is there another identifier in the login.html we can use to ensure that it's a Sercomm device?

Reply:
Unfortunately not